tests/ga/test_extension.py

```diff
@@ -1315,55 +1390,25 @@ def test_ext_handler_download_failure_permanent_ProtocolError(self, mock_add_eve
         mock_error_state.return_value = True
 
         exthandlers_handler.run()
+        exthandlers_handler.report_ext_handlers_status()
+
         event_occurrences = [kw for _, kw in mock_add_event.call_args_list if
                              "[ExtensionError] Failed to get ext handler pkgs" in kw['message']]
         self.assertEqual(1, len(event_occurrences))
         self.assertFalse(event_occurrences[0]['is_success'])
         self.assertTrue("Failed to get ext handler pkgs" in event_occurrences[0]['message'])
         self.assertTrue("ProtocolError" in event_occurrences[0]['message'])
 
-    @patch('azurelinuxagent.ga.exthandlers.add_event')
```

Comment:
some tests (or sections of tests) don't make sense anymore since the entire logic for processing extensions is short-circuited if there not a new goal state